alkira/resource_alkira_virtual_network_manager_azure.go

```diff
@@ -0,0 +1,207 @@
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Alkira Azure Virtual Network Manager.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"region": {
+				Description: "The Azure Region where the Azure Virtual Network" +
+					" Manager will be created. eg: `eastus`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"subscription_id": {
+				Description: "The ID of the Azure Subscription account in which the" +
+					" Azure Virtual Network Manager is created.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cxp": {
+				Description: "The CXP on which the virtual network manager is created.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"resource_group": {
+				Description: "The name of the Azure Resource Group in which the" +
+					" Azure Virtual Network Manager is created.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Description: "The description of the Alkira Azure Virtual Network Manager.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"credential_id": {
+				Description: "ID of credential managed by Credential Manager.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"subscriptions_in_scope": {
+				Description: "List of IDs of Azure Subscription Accounts that" +
+					" will be managed by this Azure Virtual Network Manager.",
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
+			},
+		},
+	}
+
+}
+
+func resourceVirtualNetworkManagerAzureCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	// INIT
+	api := alkira.NewVirtualNetworkManagerAzure(m.(*alkira.AlkiraClient))
+
+	request, err := generateAzureVirtualNetworkManagerRequest(d, m)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resource, _, err, _ := api.Create(request)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(string(resource.Id))
+
+	// WAITING FOR THE STATE
+	state := resource.State
+	for state != "ACTIVE" {
+		resource, _, err := api.GetById(d.Id())
+
+		if err != nil {
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "FAILED TO GET RESOURCE",
+				Detail:   fmt.Sprintf("%s", err),
+			}}
+		}
+
+		state = resource.State
+		time.Sleep(5 * time.Second)
+
+	}
+	return resourceVirtualNetworkManagerAzureRead(ctx, d, m)
+}
+
+func resourceVirtualNetworkManagerAzureUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// CHECK FOR CHANGES IN ANY FIELD EXCEPT DESCRIPTION AND RETURN ERR.
+	if d.HasChange("description") {
```

Comment:
If only description could be updated for a resource, I think we should consider this resource not upgradable at all.

Reply:
This is same as the peering gateway, we can only update the description. Otherwise we throw an error saying it is an "Invalid Update".